### ISSUE TITLE
Toolkit support for Arm64 Visual Studio

### ISSUE THREAD

**Issue by awschristou** [aws/aws-toolkit-visual-studio#302]
## _Update_ 

(2023/09/05) Support for Arm64 Visual Studio is no longer in Preview mode and is now available with version 1.43.0.0. See also [AWS Toolkit for Visual Studio adds support for Arm64 Visual Studio](https://aws.amazon.com/blogs/developer/aws-toolkit-for-visual-studio-adds-support-for-arm64-visual-studio/)

(2023/06/21) Preview support for Arm64 Visual Studio is now available from the [Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=AmazonWebServices.AWSToolkitforVisualStudio2022)! Starting with version 1.42.0.0, you can now install the Toolkit the same way as you can for x64 processors.

While the Toolkit's Arm support is in Preview mode, we would like your feedback around what is or isn't working. The Toolkit has a banner that you can use to send feedback and file issues relating to your experience.

(original issue text below)

---

The Toolkit currently has support for x86 based hardware. In November 2022, an [Arm64 version of Visual Studio](https://devblogs.microsoft.com/visualstudio/arm64-visual-studio-is-officially-here/) was released. We have started receiving feature requests for the Toolkit to support Arm64. #291 for example illustrates a use case running Visual Studio within Parallels on Arm based MacBooks.

If this is something you would find useful, you can register your interest by adding a reaction to this issue:
![image](https://user-images.githubusercontent.com/39839589/202237133-9ba55446-7251-4afd-9351-a42bd94f3953.png)

Registering your interest helps us determine what features to prioritize.

## Current Behavior

The Visual Studio extension installer (VSIX Installer) fails to install the Toolkit, and the installer log contains messages similar to:
```
Supported Products :
Microsoft.VisualStudio.xxx
Version : [17.0,18.0)
ProductArchitecture : amd64

Found installed product - Global Location
Found installed product - Visual Studio xxx 2022
VSIXInstaller.NoApplicableSKUsException: This extension is not installable on any currently installed products.
```

## Alternatives
- [AWS Toolkit for VS Code](https://aws.amazon.com/visualstudiocode/)
- [AWS Toolkit for JetBrains](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/welcome.html)

## Background
The Toolkit's feature set needs to be tested using an Arm based system in order to determine which features do and do not work out of the box. For features that do not work, we will have to figure out what changes are required in order to support Arm.

**Comment by davidscheidt** [comment]
Just wanted to let you know, it worked on Windows on ARM / Parallels / MacBook before the 'official' 17.4 release of Visual Studio 2022.  Was able to install and work with Cognito, S3, API Gateway, and Lambda, including being able to deploy my .NET project correctly

**Comment by cjsuarez22** [comment]
Any ETA about it.
Windows running on ARM architecture is becoming more common and even non-Microsoft hardware is embracing the technology. So this issue will exponentially become more urgent week after week.

**Comment by awschristou** [comment]
@cjsuarez22 I don't have an ETA for this, but it is on our roadmap to investigate running the Toolkit on ARM.

**Comment by awschristou** [comment]
Just a brief update on this - we have started investigating running the Toolkit on ARM based Visual Studio

**Comment by cjsuarez22** [comment]
Excellent. Thank you

Sent from my iPad

On Apr 12, 2023, at 19:09, Christopher Christou ***@***.***> wrote:

﻿

Just a brief update on this - we have started investigating running the Toolkit on ARM based Visual Studio

**Comment by aclawsonCSI** [comment]
I am wondering if someone might be able to help me understand this issue I am having around this. I started working in Visual Studio 2022 in January 2022 on a brand new Apple M1 MacBook Pro. I had not previously used the AWS toolkit and was pleasantly surprised when I found it in the marketplace. It downloaded and installed on my ARM MacBook and I've used it consistently until an update to visual studio wrecked my system in March. I've setup a new Windows VM and have re-established my development setup but now the AWS toolkit won't even install. My question is...why did the toolkit ever install and run correctly if it's not supposed to work on the ARM architecture? At any rate, I highly support getting this running on ARM architecture as my development workflow is highly dependent on easy deployment to Elasticbeanstalk. Thank you.

**Comment by awschristou** [comment]
We have a test build of the AWS Toolkit for Visual Studio that works with the ARM version of Visual Studio 2022. We would like your feedback around what is or isn't working in this test build! Please try it out on ARM based systems, and comment in this issue what features you used and whether or not they worked for you. This feedback will help us prepare to include ARM support in the Toolkit published through the marketplace.

The test build can be downloaded from the release assets in https://github.com/aws/aws-toolkit-visual-studio/releases/tag/TestArmSupport

**Comment by brminnick** [comment]
Awesome! I can confirm that the installation worked (which is where the current release in the Marketplace fails).

I can also confirm that Right Click -> `Publish to AWS Lambda` works.

Thanks so much for completing this!

**Comment by awschristou** [comment]
We have started including Arm64 support in the Toolkit that is published into the Marketplace. See [the original post](https://github.com/aws/aws-toolkit-visual-studio/issues/302#issue-1451940198) for more details.

**Comment by awschristou** [comment]
Support for Arm64 Visual Studio is no longer in Preview mode and is now available with version 1.43.0.0.
